cli/azd/test/functional/cli_test.go

```diff
@@ -690,3 +710,10 @@ func newTestContext(t *testing.T) (context.Context, context.CancelFunc) {
 
 	return context.WithCancel(ctx)
 }
+
+func IntegrationTest(t *testing.T) {
+	t.Helper()
+	if testing.Short() {
+		t.Skip("skipping integration tests")
+	}
```

Comment:
I wonder if we should not depend on a function like `IntegrationTest` for doing the skipping...

Did you consider moving these tests to a different package and use the go-packages to set what we want to run and what we don't want to run??

For instance, if we ever add a new test package, we will need to use `cli_test.IntegratrionTest(t)` and create a dependency between test packages.

Can we probably let the package `cli_test` to be considered as `integration` and use for any test creating resources ?

Reply:
> Did you consider moving these tests to a different package and use the go-packages to set what we want to run and what we don't want to run??

There's a circular problem here. You want to have a `GOROOT`, a common tree folder, that allows `go build` and `go test` in one command. We expect to build these integration tests whenever a code changes occur.

> For instance, if we ever add a new test package, we will need to use cli_test.IntegratrionTest(t) and create a dependency between test packages.

I think we should follow YAGNI and refactor our common code when we needed. The implementation here relies on `testing.Short()`, and thus any test can mark their test as non-applicable to short-mode. This is simply a nice wrapper around that. It seems that having a `functional` package that contains all end-to-end tests would be sufficient, and there would not be a different package.

> Can we probably let the package cli_test to be considered as integration and use for any test creating resources ?

There isn't a way to mark packages as integration test. The only alternative is to use a regex on the test names, which is what is being done currently.

Reply:
What I would like is to avoid marking tests as `short` or `not short`.
From the test perspective, I would like that to be non-relevant.

> There's a circular problem here

How so?  
I am not saying that we move those tests out of `/cli/azd`.  I am saying that we can pick what packages to run when we do `go test`.   Instead of using `go test ./...`, we can do `go test $(go list ./... | grep -v clit_test)` to test all packages but `cli_test` (without updating all tests to short or not short)

Reply:
We discussed offline and Victor's main concern is about separation of concerns between the test implementation and configuration about the test run itself. I pushed a change to extract this out to `TestMain`. Alternatively, we could consider using `-skip` once https://github.com/golang/go/issues/41583 lands.

I would like to avoid introduce a script wrapper if possible.

@ellismg for any thoughts on what might be more suitable for go.

Reply:
The `TestMain` stuff looks reasonable to me.  Like you, I would prefer this over a wrapper script or something.  Agree that moving to `-skip` in 1.20 makes sense.